srml/executive/src/lib.rs

```diff
@@ -369,8 +385,9 @@ where
 					vec![]
 				};
 
+				// TODO: maximise (fee + tip) per weight unit here.
 				TransactionValidity::Valid {
-					priority: encoded_len as TransactionPriority,
+					priority: (encoded_len as TransactionPriority) + tip_value.saturated_into::<TransactionPriority>(),
```

Comment:
I don't think this is correct. We seem to be adding a bytes length of the extrinsic and the value, but since `MakePayment` is generic, we don't really know how much each byte of `encoded_len` is worth, so it's hard to compare it directly with `Balance`.
I think `MakePayment` trait, should handle all this (maybe it requires a slight rename though):
`fn make_payment(sender, encoded_len, tip) -> Result<Priority>;`

So we make payment for both len and tip in one operation and we get the transaction priority back (or something that can be easily converted to priority).